docs/concepts/controller-manager.md

```diff
@@ -72,4 +72,54 @@ The `--event-ttl` should be larger than the `ttlNonShootEvents` or this controll
 
 ## Webhooks
 
-ℹ️ This document is incomplete and under construction.
+While the Gardener API server works with [admission plugins](./apiserver_admission_plugins.md) to validate and mutate resources belonging to Gardener related API groups, e.g. `core.gardener.cloud`, the same is needed for resources with arbitrary API groups as well.
+Therefore, the Gardener Controller Manager runs a http(s) server with the following handlers which serve as validating/mutating endpoints for [admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+### Kubeconfig Secret Validator
+
+[Malicious Kubeconfigs](https://github.com/kubernetes/kubectl/issues/697) applied by end users may cause a leakage of sensitive data.
+This handler checks if the incoming request contains a Kubernetes secret with a `.data.kubeconfig` field and denies the request if the Kubeconfig structure violates Gardener's security standards.
+
+### Namespace Validator
+
+Namespaces are the backing entities of Gardener projects in which shoot clusters objects reside.
+This validation handler protects active namespaces against premature deletion requests.
+Therefore, it denies deletion requests if a namespace still contains shoot clusters or if it belongs to a non-deleting Gardener project (project w/o `.meta.deletionTimestamp`).
+
+### Resource Size Validator
+
+Since users directly apply Kubernetes native objects to the Garden cluster, it also involves the risk of being susceptible to DoS attacks because these resources are again read by further controllers in the Gardener landscape.
```

Comment:
```suggestion
Since users directly apply Kubernetes native objects to the Garden cluster, it also involves the risk of being vulnerable to DoS attacks because these resources are read by controllers in the Gardener landscape.
```